src/math/Triangle.js

```diff
@@ -60,9 +60,8 @@ class Triangle {
 		// collinear or singular triangle
 		if ( denom === 0 ) {
 
-			// arbitrary location outside of triangle?
-			// not sure if this is the best idea, maybe should be returning undefined
-			return target.set( - 2, - 1, - 1 );
+			target.set( 0, 0, 0 );
+			return null;
```

Comment:
@WestLangley Do you think this is okay? I don't know if `0, 0, 0` is better than `- 2, - 1, - 1` but I like the idea or returning `null` to indicate an erroneous edge case.

Reply:
How about
```js
target.set( 1, 0, 0 );

return null;
```

Reply:
Can you explain you're reasoning? 1, 0, 0 is a reasonable result that could be returned by the function when using a valid triangle. I picked 0, 0, 0 because it's a value that cannot be produced in a valid scenario making it easier to see that a failure is occuring.

Reply:
You are returning `null` to indicate that the triangle is degenerate. The app can handle the returned result as it pleases.

Reply:
Against this backdrop I vote to keep the previous value. If a project specifically tested for `(- 2, - 1, - 1)` to detect an error, we do not break this usage.

Reply:
Hmm... Maybe `getInterpolation()` should also test for `null` in case the points are collinear?

I do not really like retuning nonsense values.

Reply:
>I'm just suggesting that if we're going to set the value to anything we should set [it] to 0, 0, 0

I was concerned that the other methods that call `getBarycoord()` will return nonsense if `getBarycoord()` returns nonsense.

```js
const t = new THREE.Triangle(); // degenerate
const p = new THREE.Vector3( 1, 1, 1 );

console.log( t.containsPoint( p ) ); // true
```

I think I'll support whatever you prefer to do here. :-)

Reply:
> Returning -2, -1, -1 was never a documented case.

Even if it is not documented, devs could look in the code see what's happening. But maybe I'm too overcautious here. We can try `(0,0,0)` and see if this causes any troubles.

Reply:
`(0,0,0)` will cause troubles with `containsPoint()`. A degenerate triangle will contain any point.

Maybe that is why `(-2, -1, -1)` was selected -- so `containsPoint()` will return `false`.

Reply:
`(0,0,0)` will cause troubles with `containsPoint()`. (A degenerate triangle will contain any point.)

Maybe that is why `(-2, -1, -1)` was selected -- so `containsPoint()` will return `false`.

Reply:
Um, but @gkjohnson updated the method. It should return `false` because it checks for `null` now.